### DISCUSSION TITLE
Read the text from a txt file as input

### DISCUSSION THREAD

**Discussion by Showwwwwwwww** [ggml-org/llama.cpp#3264] [Q&A]
Hi, 
I am currently working with llama.cpp on my macOS machine and attempting to use a sentence from another conversation as input continually for the llama model. Unfortunately, I have encountered difficulties in configuring the program to read the conversation text as input.

Could you possibly provide guidance or suggest a method that would allow me to successfully integrate the conversation text as input into the llama model?

**Comment by ianscrivener** [comment]
```
echo 'Please complete the following sentence:\n "Mary had a little lamb..."' > prompt.txt

./main \
  -t 8 \
  --model ./models/7B/llama-2-7b-chat.ggmlv3.q6_K.gguf \
  --file prompt.txt \
  --ctx-size 2048

```

**Reply by Showwwwwwwww** [reply]
Hi,
I can understand that this is using prompt, but it can only used before the conversation for once. I am trying to set it up such that it continuously captures text from a specific .txt file as input throughout the conversation, instead of only before the conversation begins. 

Currently, I am trying to modify the example/main.cpp file to read a specific .txt file for the next input, instead of utilizing the readline() function to obtain input from the console.

But I am wondering if this functionality can be implemented without modifying the original code.

**Reply by ianscrivener** [reply]
What about using [llama-cpp-python](https://github.com/abetlen/llama-cpp-python)? It is a python wrapper for llama.cpp
Easier (for most) than coding C++/C

**Reply by Showwwwwwwww** [reply]
Thanks for the information.  I am trying to install it, however, I meet the error message for "FileNotFoundError: Shared library with base name 'llama' not found".  Have quite confused about that.

**Reply by ianscrivener** [reply]
a few troubleshooting tips;
**1)** use python v3.10.12
**2)** force and download build llama-cpp-python;
 eg `CMAKE_ARGS="-DLLAMA_METAL=on" pip install -U llama-cpp-python --upgrade --no-cache-dir`
       `CMAKE_ARGS="-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=OpenBLAS" pip install llama-cpp-python --upgrade --no-cache-dir`
**3)** Put the gguf model in the same directory as the python script

**Reply by Showwwwwwwww** [reply]
It still does nots works for me. It said that all the requirement already satisfied.  

The error message still continue with:

Traceback (most recent call last):
  File "<frozen runpy>", line 189, in _run_module_as_main
  File "<frozen runpy>", line 112, in _get_module_details
  File "/Users/chen/Desktop/robot/robot_research/llama2/llama-cpp-python/llama_cpp/__init__.py", line 1, in <module>
    from .llama_cpp import *
  File "/Users/chen/Desktop/robot/robot_research/llama2/llama-cpp-python/llama_cpp/llama_cpp.py", line 80, in <module>
    _lib = _load_shared_library(_lib_base_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/chen/Desktop/robot/robot_research/llama2/llama-cpp-python/llama_cpp/llama_cpp.py", line 71, in _load_shared_library
    raise FileNotFoundError(
FileNotFoundError: Shared library with base name 'llama' not found

**Reply by ianscrivener** [reply]
As this is a llama-cpp-python issue you are better to post this on the [llama-cpp-python](https://github.com/abetlen/llama-cpp-python) Dicussion - they may have had this issue before.
